connection/cluster/ClusterUserManager.java

```diff
@@ -73,16 +73,70 @@ public void create(String username, String password) {
         failsafeTask.runPrimaryReplica();
     }
 
+    @Override
+    public void delete(String username) {
+        ClusterClient.FailsafeTask<Void> failsafeTask = client.createFailsafeTask(
+                SYSTEM_DB,
+                parameter -> {
+                    parameter.client().stub().usersDelete(deleteReq(username));
+                    return null;
+                }
+        );
+        failsafeTask.runPrimaryReplica();
+    }
+
     @Override
     public Set<User> all() {
         ClusterClient.FailsafeTask<Set<User>> failsafeTask = client.createFailsafeTask(
                 SYSTEM_DB,
-                (parameter) ->
-                        parameter.client().stub()
-                                .usersAll(allReq()).getNamesList().stream()
-                                .map(name -> new ClusterUser(client, name))
-                                .collect(Collectors.toSet())
+                (parameter) -> {
+                    Set<User> users = new HashSet<>();
+                    List<ClusterUserProto.User> protoUsers = parameter.client().stub().usersAll(allReq()).getUsersList();
+
+                    for (ClusterUserProto.User user: protoUsers) {
+                        if (user.getPasswordExpiryOptCase() ==
+                                ClusterUserProto.User.PasswordExpiryOptCase.PASSWORDEXPIRYOPT_NOT_SET) {
+                            users.add(new ClusterUser(client, user.getUsername(), Optional.empty()));
+                        }
+                        else {
+                            users.add(new ClusterUser(client, user.getUsername(), Optional.of(user.getPasswordExpiryDays())));
+                        }
+                    }
+
+                    return users;
+                }
+
+        );
+        return failsafeTask.runPrimaryReplica();
+    }
+
+    @Override
+    public User get(String username) {
+        ClusterClient.FailsafeTask<User> failsafeTask = client.createFailsafeTask(
+                SYSTEM_DB,
+                (parameter) -> {
+                    com.vaticle.typedb.protocol.ClusterUserProto.User user = parameter.client().stub().usersGet(getReq(username)).getUser();
+                    if (user.getPasswordExpiryOptCase() ==
+                            ClusterUserProto.User.PasswordExpiryOptCase.PASSWORDEXPIRYOPT_NOT_SET) {
+                        return new ClusterUser(client, user.getUsername(), Optional.empty());
+                    }
+                    else {
+                        return new ClusterUser(client, user.getUsername(), Optional.of(user.getPasswordExpiryDays()));
+                    }
```

Comment:
Ok at this point - notice we have repeated code, then we realise there is a better domain to put the construction of a `ClusterUser` - the `ClusterUser` itself!

Reply:
Agreed, implemented.